pkg/app/piped/executor/ecs/rollback.go

```diff
@@ -158,8 +158,21 @@ func rollback(ctx context.Context, in *executor.Input, platformProviderName stri
 			return false
 		}
 
+		currListenerRuleArns, err := client.GetListenerRuleArns(ctx, currListenerArns)
+		if err != nil {
+			in.LogPersister.Errorf("Failed to get current active listener rule: %v", err)
+			return false
+		}
+
+		if len(currListenerRuleArns) > 0 {
+			if err := client.ModifyRules(ctx, currListenerRuleArns, routingTrafficCfg); err != nil {
+				in.LogPersister.Errorf("Failed to routing traffic to PRIMARY/CANARY variants: %v", err)
+				return false
+			}
```

Comment:
Should return after this line, or `ModifyListeners` will be triggered and override every rules that we want to keep 👀

Reply:
[0b01246](https://github.com/pipe-cd/pipecd/pull/4668/commits/0b01246eabf38c6c9035c755e7d59727001ae5e0)
Sorry about that oversight. I've made the correction to ensure ModifyListeners doesn't override our rules